coalib/output/Logging.py

```diff
@@ -0,0 +1,84 @@
+def configure_logging():
+    """
+    Configures the logging with hard coded dictionary.
+    """
+    import sys
+    import logging.config
+
+    logging.config.dictConfig({
+        'version': 1,
+        'handlers': {
+            'colored': {
+                'class': 'logging.StreamHandler',
+                'formatter': 'colored',
+                'stream': sys.stderr
+            }
+        },
+        'root': {
+            'level': 'DEBUG',
+            'handlers': ['colored']
+        },
+        'formatters': {
+            'colored': {
+                '()': 'colorlog.ColoredFormatter',
+                'format': '%(log_color)s[%(levelname)s]%(reset)s[%(asctime)s]'
+                          ' %(message)s',
+                'datefmt': '%X',
+                'log_colors': {
+                    'ERROR': 'red',
+                    'WARNING': 'yellow',
+                    'INFO': 'blue',
+                    'DEBUG': 'green'
+                }
+            }
+        }
+    })
+
+
+def configure_json_logging():
+    """
+    Configures logging as a JSON objects.
+    Returns a StringIO() that captures the logs as json.
+    """
+    import logging.config
+    import io
+
+    stream = io.StringIO()
+
+    logging.config.dictConfig({
+        'version': 1,
+        'handlers': {
+            'json': {
+                'class': 'logging.StreamHandler',
+                'formatter': 'json',
+                'stream': stream
+            }
+        },
+        'root': {
+            'level': 'DEBUG',
+            'handlers': ['json']
+        },
+        'formatters': {
+            'json': {
+                '()': 'logging.Formatter',
+                'format': '{ "message": "%(message)s",'
+                          ' "level": "%(levelname)s",'
+                          ' "time": "%(asctime)s" }|',
+                'datefmt': '%X',
+            }
+        }
+    })
+    return stream
+
+
+def parse_json_log(stream):
+    """
+    Parses A Stream of JSON Handler
+    Returns a list of JSON object for logs
+    """
+    import json
+
+    logs = list(map(str.strip, stream.getvalue().split('|')))
+    logs = [json.loads(log, strict=False) for log in logs if log]
+
+    return logs
```

Comment:
yep, there *must* be a better way, writing a custom formatter that gets the actual log objects with the data instead of formatting it and reparsing it back